File: quizzes/quizzesV2/quizzes10/08-problem.ts
export {};

const fibLogger = (num: number): number[] => {
    if (!Number.isInteger(num) || num < 2) return [];
    const fib: number[] = [0, 1];
    if (num === 2) return fib;

    for (let i = 2; i < num; i++) {
        fib.push(fib[i - 1] + fib[i - 2]);
    }

    return fib;
};

const fibFinder = (num: number): number => {
    let a: number = 0;
    if (!Number.isInteger(num) || num < 2) return a;
    let b: number = 1;
    if (num === 2) return b;

    for (let i = 2; i < num; i++) {
        [a, b] = [b, a + b];
    }

    return b;
};

console.log(JSON.stringify({ data: fibLogger(10) }, null, 2));
console.log(JSON.stringify({ data: fibFinder(10) }, null, 2));

File: bins/binsV2/bin/bins5/08-problem.rb
# frozen_string_literal: true

require 'json'

def fib_logger(num)
  return [ 0 ] unless num.is_a?(Integer) && num >= 2

  fib = [ 0, 1 ]
  return fib if num === 2

  (2...num).each do |i|
    fib << fib[i - 1] + fib[i - 2]
  end

  fib
end

def fib_finder(num)
  a, b = 0, 1
  return a unless num.is_a?(Integer) && num >= 2
  return b if num === 2

  (2...num).each do |i|
    a, b = b, a + b
  end

  b
end

puts JSON.pretty_generate({ data: fib_logger(10) })
puts JSON.pretty_generate({ data: fib_finder(10) })
